s=$1
c=$2

rm -r /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$c"
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"/
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$c"/

for t in deletion50 insertion50 inversion seg_deletion seg_duplication
do
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$c"/Variants/"$c"-v-Kestrel100KBhap.SV.kestCoord."$t".bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap.kestCoord."$t".bed

KEST_COORDS=`cat /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap.kestCoord."$t".bed | perl -pe 's#\t#__#g'`
rm /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"/"$s"-v-"$c"_Kestrel100KBhap."$s"-coords."$t".bed
rm /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"-coords."$t".bed
for p in $KEST_COORDS
do
KEST_SCAFF=`echo $p | perl -pe 's#__#\t#g' | awk '{print $1}'`
KEST_POS1=`echo $p | perl -pe 's#__#\t#g' | awk '{print $2}'`
KEST_POS2=`echo $p | perl -pe 's#__#\t#g' | awk '{print $3}'`
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk '$15~/_1/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $2<=kest_pos1{QScaff=$15; QPos1=$4; next}$14==kest_scaff && $1>=kest_pos2 && $15==QScaff{QPos2=$3}$14==kest_scaff && $1>=kest_pos2 && $15!=QScaff{exit}QPos1<QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos1 OFS QPos2; exit}QPos1>QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos2 OFS QPos1; exit}' >> /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"/"$s"-v-"$c"_Kestrel100KBhap."$s"-coords."$t".bed
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk '$15~/_2/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $2<=kest_pos1{QScaff=$15; QPos1=$4; next}$14==kest_scaff && $1>=kest_pos2 && $15==QScaff{QPos2=$3}$14==kest_scaff && $1>=kest_pos2 && $15!=QScaff{exit}QPos1<QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos1 OFS QPos2; exit}QPos1>QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos2 OFS QPos1; exit}' >> /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"/"$s"-v-"$c"_Kestrel100KBhap."$s"-coords."$t".bed

grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$c"-v-Kestrel100KBhap.coords | awk '$15~/_1/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $2<=kest_pos1{QScaff=$15; QPos1=$4; next}$14==kest_scaff && $1>=kest_pos2 && $15==QScaff{QPos2=$3}$14==kest_scaff && $1>=kest_pos2 && $15!=QScaff{exit}QPos1<QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos1 OFS QPos2; exit}QPos1>QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos2 OFS QPos1; exit}' >> /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"-coords."$t".bed
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$c"-v-Kestrel100KBhap.coords | awk '$15~/_2/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $2<=kest_pos1{QScaff=$15; QPos1=$4; next}$14==kest_scaff && $1>=kest_pos2 && $15==QScaff{QPos2=$3}$14==kest_scaff && $1>=kest_pos2 && $15!=QScaff{exit}QPos1<QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos1 OFS QPos2; exit}QPos1>QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos2 OFS QPos1; exit}' >> /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"-coords."$t".bed
done

bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"/"$s"-v-"$c"_Kestrel100KBhap."$s"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$s"coord."$t"-rep50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"/"$s"-v-"$c"_Kestrel100KBhap."$s"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.simp.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$s"coord."$t"-simp50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"/"$s"-v-"$c"_Kestrel100KBhap."$s"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.DNA.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$s"coord."$t"-DNA50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"/"$s"-v-"$c"_Kestrel100KBhap."$s"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.LTR.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$s"coord."$t"-LTR50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"/"$s"-v-"$c"_Kestrel100KBhap."$s"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.LINE.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$s"coord."$t"-LINE50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"/"$s"-v-"$c"_Kestrel100KBhap."$s"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.SINE.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$s"coord."$t"-SINE50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$s"/"$s"-v-"$c"_Kestrel100KBhap."$s"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.unknown.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$s"coord."$t"-unknown50.bed

bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$c"/RepeatBeds/"$c"_16KBph2.diploid.fasta.rep.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"coord."$t"-rep50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$c"/RepeatBeds/"$c"_16KBph2.diploid.fasta.rep.simp.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"coord."$t"-simp50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$c"/RepeatBeds/"$c"_16KBph2.diploid.fasta.rep.DNA.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"coord."$t"-DNA50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$c"/RepeatBeds/"$c"_16KBph2.diploid.fasta.rep.LTR.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"coord."$t"-LTR50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$c"/RepeatBeds/"$c"_16KBph2.diploid.fasta.rep.LINE.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"coord."$t"-LINE50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$c"/RepeatBeds/"$c"_16KBph2.diploid.fasta.rep.SINE.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"coord."$t"-SINE50.bed
bedtools subtract -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds/"$s"-v-"$c"/"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"-coords."$t".bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$c"/RepeatBeds/"$c"_16KBph2.diploid.fasta.rep.unknown.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds/"$s"-v-"$c"/"$s"-v-"$c"_Kestrel100KBhap."$c"coord."$t"-unknown50.bed
done
