#!/bin/bash
#SBATCH --time=168:00:00
#SBATCH --mem=60GB
#SBATCH -p serial
# Set number of nodes to run
# Set number of tasks to run
#SBATCH --ntasks=14
# Set number of cores per task (default is 1)
#SBATCH --cpus-per-task=1
# Output and error files
#SBATCH -o job.%J.out

module purge
module load perl
module load gencore/1
module load gencore_variant_detection/1.0

rm -r /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/RepeatBeds
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/LongRanger
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/GenomeBeds
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/Intersect/Rep-SVBeds


#Extract bed file of break points from MUMmer kestrel align ".coords" file
for s in A D B FpBSx77 C E F Kestrel
do
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SV-TranscriptBeds/
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-TranscriptBeds/
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-TranscriptBeds/
mkdir /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/

#Extract Repeats BedFile
grep -P "[_0-9+C]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeats/16KBph2Diploid/RepeatMasker/rmBLAST_Ultimate/Annotations/"$s"_16KBph2.diploid.fasta.out | awk 'BEGIN{OFS="\t"}{print $5 OFS $6 OFS $7}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.bed &
grep -P "[_0-9+C]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeats/16KBph2Diploid/RepeatMasker/rmBLAST_Ultimate/Annotations/"$s"_16KBph2.diploid.fasta.out | awk 'BEGIN{OFS="\t"}$11~/Simple_repeat/{print $5 OFS $6 OFS $7}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.simp.bed &
grep -P "[_0-9+C]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeats/16KBph2Diploid/RepeatMasker/rmBLAST_Ultimate/Annotations/"$s"_16KBph2.diploid.fasta.out | awk 'BEGIN{OFS="\t"}$11~/DNA/{print $5 OFS $6 OFS $7}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.DNA.bed &
grep -P "[_0-9+C]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeats/16KBph2Diploid/RepeatMasker/rmBLAST_Ultimate/Annotations/"$s"_16KBph2.diploid.fasta.out | awk 'BEGIN{OFS="\t"}$11~/LTR/{print $5 OFS $6 OFS $7}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.LTR.bed &
grep -P "[_0-9+C]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeats/16KBph2Diploid/RepeatMasker/rmBLAST_Ultimate/Annotations/"$s"_16KBph2.diploid.fasta.out | awk 'BEGIN{OFS="\t"}$11~/LINE/{print $5 OFS $6 OFS $7}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.LINE.bed &
grep -P "[_0-9+C]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeats/16KBph2Diploid/RepeatMasker/rmBLAST_Ultimate/Annotations/"$s"_16KBph2.diploid.fasta.out | awk 'BEGIN{OFS="\t"}$11~/SINE/{print $5 OFS $6 OFS $7}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.SINE.bed &
grep -P "[_0-9+C]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeats/16KBph2Diploid/RepeatMasker/rmBLAST_Ultimate/Annotations/"$s"_16KBph2.diploid.fasta.out | awk 'BEGIN{OFS="\t"}$11~/Unknown/{print $5 OFS $6 OFS $7}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.unknown.bed &
wait

grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"}$3<$4{print $15 OFS $3 OFS $4}$3>$4{print $15 OFS $4 OFS $3}' | sort -k1,1V -k2,2n > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"-v-Kestrel100KBhap.align.bed
awk '{print $1}' /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"-v-Kestrel100KBhap.align.bed | sort | uniq | grep -f - /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/"$s"_16KBph2.diploid.fasta.fai | awk 'BEGIN{OFS=FS="\t"}{print $1, $2}' | sort -k1,1V -k2,2n > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"_16KBph2.diploid.genome
bedtools complement -i /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"-v-Kestrel100KBhap.align.bed -g /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"_16KBph2.diploid.genome | awk '{Size=$3 - $2}Size>50 && $2>50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.temp.bed
SCAFF_SIZES=`cat /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"_16KBph2.diploid.genome | perl -pe 's#\t#__#'`
rm /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed
for v in $SCAFF_SIZES
do
SCAFF=`echo $v | perl -pe 's#__#\t#g' | awk '{print $1}'`
SIZE=`echo $v | perl -pe 's#__#\t#g' | awk '{print $2}'`
awk -v scaff=$SCAFF -v size=$SIZE 'BEGIN{size=size -50}$1==scaff && $3<size{print $0}' /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.temp.bed >> /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed
done
rm /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.temp.bed


grep -v "#" /scratch/jw5478/FalconProject/Projects/RefGenome8/LongrangerOut/SelfAlign/16KBph2Haploid/"$s"_16KBph2-HaploidLR/outs/large_sv_calls.bedpe | grep "ZS=HET" | awk 'BEGIN{OFS=FS="\t"}$1==$4{print $1 OFS $2 OFS $5}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/LongRanger/"$s"-v-"$s"_LR.SV.bed
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/LongRanger/"$s"-v-"$s"_LR.SV.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SVmum-v-lr.bed
bedtools intersect -F 0.5 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/LongRanger/"$s"-v-"$s"_LR.SV.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SVmum-v-lr50-50.bed

bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.simp.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel00KBhap.SV-rep.simp.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.DNA.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.DNA.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.LTR.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.LTR.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.LINE.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.LINE.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.SINE.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.SINE.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.unknown.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.unknown.bed &

bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/LongRanger/"$s"-v-"$s"_LR.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.lrSV-v-rep.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/LongRanger/"$s"-v-"$s"_LR.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel00KBhap.SV-rep.simp.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.lrSV-v-rep.simp.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/LongRanger/"$s"-v-"$s"_LR.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.DNA.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.lrSV-v-rep.DNA.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/LongRanger/"$s"-v-"$s"_LR.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.LTR.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.lrSV-v-rep.LTR.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/LongRanger/"$s"-v-"$s"_LR.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.LINE.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.lrSV-v-rep.LINE.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/LongRanger/"$s"-v-"$s"_LR.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.SINE.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.lrSV-v-rep.SINE.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/LongRanger/"$s"-v-"$s"_LR.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.SV-rep.unknown.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"-v-Kestrel100KBhap.lrSV-v-rep.unknown.bed &
wait

grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"}$1<$2{print $14 OFS $1 OFS $2}$1>$2{print $14 OFS $2 OFS $1}' | sort -k1,1V -k2,2n > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"-v-Kestrel100KBhap.align.kestCoordbedAlign.bed
awk '{print $1}' /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"-v-Kestrel100KBhap.align.kestCoordbedAlign.bed | sort | uniq | grep -f - /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/Kestrel_16KBph2.100KB.haploid.fasta.fai | awk 'BEGIN{OFS=FS="\t"}{print $1, $2}' | sort -k1,1V -k2,2n > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"-v-Kestrel100KBhap.align.kestCoordbed.genome
bedtools complement -i /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"-v-Kestrel100KBhap.align.kestCoordbedAlign.bed -g /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"-v-Kestrel100KBhap.align.kestCoordbed.genome | awk '{Size=$3 - $2}Size>50 && $2>50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.temp.bed
SCAFF_SIZES=`cat /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Alignment/"$s"-v-Kestrel100KBhap.align.kestCoordbed.genome | perl -pe 's#\t#__#'`
rm /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.bed
for v in $SCAFF_SIZES
do
SCAFF=`echo $v | perl -pe 's#__#\t#g' | awk '{print $1}'`
SIZE=`echo $v | perl -pe 's#__#\t#g' | awk '{print $2}'`
awk -v scaff=$SCAFF -v size=$SIZE 'BEGIN{size=size -50}$1==scaff && $3<size{print $0}' /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.temp.bed >> /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.bed
done
rm /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.temp.bed

#Coordinates on Kestrel Alignment of Deletion relative to Kestrel
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"}{test_qscaff_1=q_scaff_1; gsub("_1", "_2", test_qscaff_1); test_qscaff_2=q_scaff_2; gsub("_2", "_1", test_qscaff_2)}$14==kest_scaff_1 &&  $1<=kest_pos2_1 && $2<=kest_pos2_1 && test_qscaff_1!=$15{kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_2 &&  $1<=kest_pos2_2 && $2<=kest_pos2_2 && test_qscaff_2!=$15{kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1>kest_pos2_1+50 && $3==q_pos2_1+1 && $13==orient_1 && $13>0 && $15~/_1/{print $14 OFS kest_pos2_1 OFS $1; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1>kest_pos2_1+50 && $3==q_pos2_1-1 && $13==orient_1 && $13<0 && $15~/_1/{print $14 OFS kest_pos2_1 OFS $1; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>kest_pos2_2+50 && $3==q_pos2_2+1 && $13==orient_2 && $13>0 && $15~/_2/{print $14 OFS kest_pos2_2 OFS $1; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1>kest_pos2_1+50 && $3==q_pos2_1+1 && $13==orient_1 && $13>0 && $15~/_1/{print $14 OFS kest_pos2_1 OFS $1; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1>kest_pos2_1+50 && $3==q_pos2_1-1 && $13==orient_1 && $13<0 && $15~/_1/{print $14 OFS kest_pos2_1 OFS $1; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>kest_pos2_2+50 && $3==q_pos2_2+1 && $13==orient_2 && $13>0 && $15~/_2/{print $14 OFS kest_pos2_2 OFS $1; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>kest_pos2_2+50 && $3==q_pos2_2-1 && $13==orient_2 && $13<0 && $15~/_2/{print $14 OFS kest_pos2_2 OFS $1; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$15~/_1/{kest_scaff_1=$14; q_scaff_1=$15; kest_pos1_1=$1; kest_pos2_1=$2; q_pos1_1=$3; q_pos2_1=$4; orient_1=$13}$15~/_2/{kest_scaff_2=$14; q_scaff_2=$15; kest_pos1_2=$1; kest_pos2_2=$2; q_pos1_2=$3; q_pos2_2=$4; orient_2=$13}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' | awk '{Size=$3 - $2}Size>50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.deletion50.bed &

#Coordinates on Kestrel Alignment of Insertion relative to Kestrel: break points in kestrel alignments were insertions took place relative to kestrel
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"}{test_qscaff_1=q_scaff_1; gsub("_1", "_2", test_qscaff_1); test_qscaff_2=q_scaff_2; gsub("_2", "_1", test_qscaff_2)}$14==kest_scaff_1 &&  $1<=kest_pos2_1 && $2<=kest_pos2_1 && test_qscaff_1!=$15{kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_2 &&  $1<=kest_pos2_2 && $2<=kest_pos2_2 && test_qscaff_2!=$15{kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1==kest_pos2_1+1 && $3>q_pos2_1+50 && $13==orient_1 && $13>0 && $15~/_1/{print $14 OFS kest_pos2_1 OFS $1; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1==kest_pos2_1+1 && $3>q_pos2_1-50 && $13==orient_1 && $13<0 && $15~/_1/{print $14 OFS kest_pos2_1 OFS $1; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1==kest_pos2_2+1 && $3>q_pos2_2+50 && $13==orient_2 && $13>0 && $15~/_2/{print $14 OFS kest_pos2_2 OFS $1; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1==kest_pos2_2+1 && $3>q_pos2_2-50 && $13==orient_2 && $13<0 && $15~/_2/{print $14 OFS kest_pos2_2 OFS $1; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$15~/_1/{kest_scaff_1=$14; q_scaff_1=$15; kest_pos1_1=$1; kest_pos2_1=$2; q_pos1_1=$3; q_pos2_1=$4; orient_1=$13}$15~/_2/{kest_scaff_2=$14; q_scaff_2=$15; kest_pos1_2=$1; kest_pos2_2=$2; q_pos1_2=$3; q_pos2_2=$4; orient_2=$13}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.insertion50.bed &

#Coordinates on Kestrel Alignment of Inversion relative to Kestrel
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"; orient_1=0; orient_2=0}{test_qscaff_1=q_scaff_1; gsub("_1", "_2", test_qscaff_1); test_qscaff_2=q_scaff_2; gsub("_2", "_1", test_qscaff_2)}$14==kest_scaff_1 &&  $1<=kest_pos2_1 && $2<=kest_pos2_1 && test_qscaff_1!=$15{next}$14==kest_scaff_2 &&  $1<=kest_pos2_2 && $2<=kest_pos2_2 && test_qscaff_2!=$15{next}$14!=kest_scaff_1 || $15!=q_scaff_1 && $15~/_1/{orient_1=0}orient_1>0{orient_sign_1=1}orient_1<0{orient_sign_1=-1}$14!=kest_scaff_2 || $15!=q_scaff_2 && $15~/_2/{orient_2=0}orient_2>0{orient_sign_2=1}orient_2<0{orient_sign_2=-1}$14==kest_scaff_1 && $15==q_scaff_1 && $13!=orient_sign_1 && $15~/_1/{print $14 OFS $1 OFS $2; orient_1=orient_1+$13; next}$14==kest_scaff_2 && $15==q_scaff_2 && $13!=orient_sign_2 && $15~/_2/{print $14 OFS $1 OFS $2; orient_2=orient_2+$13; next}$15~/_1/{kest_scaff_1=$14; q_scaff_1=$15; kest_pos1_1=$1; kest_pos2_1=$2; q_pos1_1=$3; q_pos2_1=$4; orient_1=orient_1+$13}$15~/_2/{kest_scaff_2=$14; q_scaff_2=$15; kest_pos1_2=$1; kest_pos2_2=$2; q_pos1_2=$3; q_pos2_2=$4; orient_2=orient_2+$13}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' | awk '{Size=$3 - $2}Size>50{print $0}' | awk '{Size=$3 - $2}Size>50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.inversion.bed &

###Coordinates in Kestrel Genome
#Coordinates on Kestrel Alignment of segmental deletion relative to Kestrel: location of duplicated region in kestrel
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"}$14==kest_scaff_1 && $15==q_scaff_1 && $1>kest_pos2_1 && $3>=q_pos1_1 && $4<=q_pos2_1 && $13>0 && $15~/_1/{print $14 OFS $3 OFS $4; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1>kest_pos2_1 && $3<=q_pos1_1 && $4>=q_pos2_1 && $13<0 && $15~/_1/{print $14 OFS $3 OFS $4; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>kest_pos2_2 && $3>=q_pos1_2 && $4<=q_pos2_2 && $13>0 && $15~/_2/{print $14 OFS $3 OFS $4; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>kest_pos2_2 && $3<=q_pos1_2 && $4>=q_pos2_2 && $13<0 && $15~/_2/{print $14 OFS $3 OFS $4; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; next}$15~/_1/{kest_scaff_1=$14; q_scaff_1=$15; kest_pos1_1=$1; kest_pos2_1=$2; q_pos1_1=$3; q_pos2_1=$4; orient_1=$13}$15~/_2/{kest_scaff_2=$14; q_scaff_2=$15; kest_pos1_2=$1; kest_pos2_2=$2; q_pos1_2=$3; q_pos2_2=$4; orient_2=$13}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' | awk '{Size=$3 - $2}Size>50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_deletion.bed &

#Coordinates on Kestrel Alignment of segmental duplication (insertion) relative to Kestrel
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"}$14==kest_scaff_1 && $15==q_scaff_1 && $1>=kest_pos1_1 && $2<=kest_pos2_1 && $4>q_pos2_1 && $13>0 && $15~/_1/{print $14 OFS $1 OFS $2; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1>=kest_pos1_1 && $2<=kest_pos2_1 && $4<q_pos2_1 && $13<0 && $15~/_1/{print $14 OFS $1 OFS $2; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>=kest_pos1_2 && $2<=kest_pos2_2 && $4>q_pos2_2 && $13>0 && $15~/_2/{print $14 OFS $1 OFS $2; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>=kest_pos1_2 && $2<=kest_pos2_2 && $4<q_pos2_2 && $13<0 && $15~/_2/{print $14 OFS $1 OFS $2; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; next}$15~/_1/{kest_scaff_1=$14; q_scaff_1=$15; kest_pos1_1=$1; kest_pos2_1=$2; q_pos1_1=$3; q_pos2_1=$4; orient_1=$13}$15~/_2/{kest_scaff_2=$14; q_scaff_2=$15; kest_pos1_2=$1; kest_pos2_2=$2; q_pos1_2=$3; q_pos2_2=$4; orient_2=$13}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_duplication.bed &


###Coordinates in $s Genome
#Coordinates on $s genome of Deletion relative to Kestrel
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"}{test_qscaff_1=q_scaff_1; gsub("_1", "_2", test_qscaff_1); test_qscaff_2=q_scaff_2; gsub("_2", "_1", test_qscaff_2)}$14==kest_scaff_1 &&  $1<=kest_pos2_1 && $2<=kest_pos2_1 && test_qscaff_1!=$15{kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_2 &&  $1<=kest_pos2_2 && $2<=kest_pos2_2 && test_qscaff_2!=$15{kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1>kest_pos2_1+50 && $3==q_pos2_1+1 && $13==orient_1 && $13>0 && $15~/_1/{print $15 OFS q_pos2_1 OFS $3; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1>kest_pos2_1+50 && $3==q_pos2_1-1 && $13==orient_1 && $13<0 && $15~/_1/{print $15 OFS q_pos2_1 OFS $3; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>kest_pos2_2+50 && $3==q_pos2_2+1 && $13==orient_2 && $13>0 && $15~/_2/{print $15 OFS q_pos2_2 OFS $3; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>kest_pos2_2+50 && $3==q_pos2_2-1 && $13==orient_2 && $13<0 && $15~/_2/{print $15 OFS q_pos2_2 OFS $3; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$15~/_1/{kest_scaff_1=$14; q_scaff_1=$15; kest_pos1_1=$1; kest_pos2_1=$2; q_pos1_1=$3; q_pos2_1=$4; orient_1=$13}$15~/_2/{kest_scaff_2=$14; q_scaff_2=$15; kest_pos1_2=$1; kest_pos2_2=$2; q_pos1_2=$3; q_pos2_2=$4; orient_2=$13}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV."$s"coord.deletion50.bed &

#Coordinates on $s genome Alignment of Insertion relative to Kestrel
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"}{test_qscaff_1=q_scaff_1; gsub("_1", "_2", test_qscaff_1); test_qscaff_2=q_scaff_2; gsub("_2", "_1", test_qscaff_2)}$14==kest_scaff_1 &&  $1<=kest_pos2_1 && $2<=kest_pos2_1 && test_qscaff_1!=$15{kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_2 &&  $1<=kest_pos2_2 && $2<=kest_pos2_2 && test_qscaff_2!=$15{kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1==kest_pos2_1+1 && $3>q_pos2_1+50 && $13==orient_1 && $13>0 && $15~/_1/{print $15 OFS q_pos2_1 OFS $3; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1==kest_pos2_1+1 && $3>q_pos2_1-50 && $13==orient_1 && $13<0 && $15~/_1/{print $15 OFS q_pos2_1 OFS $3; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1==kest_pos2_2+1 && $3>q_pos2_2+50 && $13==orient_2 && $13>0 && $15~/_2/{print $15 OFS q_pos2_2 OFS $3; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1==kest_pos2_2+1 && $3>q_pos2_2-50 && $13==orient_2 && $13<0 && $15~/_2/{print $15 OFS q_pos2_2 OFS $3; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$15~/_1/{kest_scaff_1=$14; q_scaff_1=$15; kest_pos1_1=$1; kest_pos2_1=$2; q_pos1_1=$3; q_pos2_1=$4; orient_1=$13}$15~/_2/{kest_scaff_2=$14; q_scaff_2=$15; kest_pos1_2=$1; kest_pos2_2=$2; q_pos1_2=$3; q_pos2_2=$4; orient_2=$13}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' | awk '{Size=$3 - $2}Size>50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV."$s"coord.insertion50.bed &

#Coordinates on $s genome of Inversion relative to Kestrel
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"; orient_1=0; orient_2=0}{test_qscaff_1=q_scaff_1; gsub("_1", "_2", test_qscaff_1); test_qscaff_2=q_scaff_2; gsub("_2", "_1", test_qscaff_2)}$14==kest_scaff_1 &&  $1<=kest_pos2_1 && $2<=kest_pos2_1 && test_qscaff_1!=$15{next}$14!=kest_scaff_1 || $15!=q_scaff_1 && $15~/_1/{orient_1=0}orient_1>0{orient_sign_1=1}orient_1<0{orient_sign_1=-1}$14!=kest_scaff_2 || $15!=q_scaff_1 && $15~/_1/{orient_2=0}orient_2>0{orient_sign_2=1}orient_2<0{orient_sign_2=-1}$14==kest_scaff_1 && $15==q_scaff_1 && $13!=orient_sign_1 && $15~/_1/{print $15 OFS $3 OFS $4; orient_1=orient_1+$13; next}$14==kest_scaff_2 && $15==q_scaff_2 && $13!=orient_sign_2 && $15~/_2/{print $15 OFS $3 OFS $4; orient_2=orient_2+$13; next}$15~/_1/{kest_scaff_1=$14; q_scaff_1=$15; kest_pos1_1=$1; kest_pos2_1=$2; q_pos1_1=$3; q_pos2_1=$4; orient_1=orient_1+$13}$15~/_2/{kest_scaff_2=$14; q_scaff_2=$15; kest_pos1_2=$1; kest_pos2_2=$2; q_pos1_2=$3; q_pos2_2=$4; orient_2=orient_2+$13}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' | awk '{Size=$3 - $2}Size>50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV."$s"coord.inversion.bed &

#Coordinates on $s genome Alignment of segmental deletion relative to Kestrel: provides region that is duplicated in Kestrel
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"}$14==kest_scaff_1 && $15==q_scaff_1 && $1>kest_pos2_1 && $3>=q_pos1_1 && $4<=q_pos2_1 && $13>0 && $15~/_1/{print $15 OFS q_pos2_1 OFS $3; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1>kest_pos2_1 && $3<=q_pos1_1 && $4>=q_pos2_1 && $13<0 && $15~/_1/{print $15 OFS q_pos2_1 OFS $3; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; orient_1=0; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>kest_pos2_2 && $3>=q_pos1_2 && $4<=q_pos2_2 && $13>0 && $15~/_2/{print $15 OFS q_pos2_2 OFS $3; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>kest_pos2_2 && $3<=q_pos1_2 && $4>=q_pos2_2 && $13<0 && $15~/_2/{print $15 OFS q_pos2_2 OFS $3; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; orient_2=0; next}$15~/_1/{kest_scaff_1=$14; q_scaff_1=$15; kest_pos1_1=$1; kest_pos2_1=$2; q_pos1_1=$3; q_pos2_1=$4; orient_1=$13}$15~/_2/{kest_scaff_2=$14; q_scaff_2=$15; kest_pos1_2=$1; kest_pos2_2=$2; q_pos1_2=$3; q_pos2_2=$4; orient_2=$13}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV."$s"coord.seg_deletion.bed &

#Coordinates on $s genome Alignment of segmental duplication relative to Kestrel
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$s"-v-Kestrel100KBhap.coords | awk 'BEGIN{OFS=FS="\t"}$14==kest_scaff_1 && $15==q_scaff_1 && $1>=kest_pos1_1 && $2<=kest_pos2_1 && $4>q_pos2_1 && $13>0 && $15~/_1/{print $15 OFS $3 OFS $4; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; next}$14==kest_scaff_1 && $15==q_scaff_1 && $1>=kest_pos1_1 && $2<=kest_pos2_1 && $4<q_pos2_1 && $13<0 && $15~/_1/{print $15 OFS $3 OFS $4; kest_scaff_1=""; q_scaff_1=""; kest_pos1_1=""; kest_pos2_1=""; q_pos1_1=""; q_pos2_1=""; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>=kest_pos1_2 && $2<=kest_pos2_2 && $4>q_pos2_2 && $13>0 && $15~/_2/{print $15 OFS $3 OFS $4; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; next}$14==kest_scaff_2 && $15==q_scaff_2 && $1>=kest_pos1_2 && $2<=kest_pos2_2 && $4<q_pos2_2 && $13<0 && $15~/_2/{print $15 OFS $3 OFS $4; kest_scaff_2=""; q_scaff_2=""; kest_pos1_2=""; kest_pos2_2=""; q_pos1_2=""; q_pos2_2=""; next}$15~/_1/{kest_scaff_1=$14; q_scaff_1=$15; kest_pos1_1=$1; kest_pos2_1=$2; q_pos1_1=$3; q_pos2_1=$4; orient_1=$13}$15~/_2/{kest_scaff_2=$14; q_scaff_2=$15; kest_pos1_2=$1; kest_pos2_2=$2; q_pos1_2=$3; q_pos2_2=$4; orient_2=$13}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' | awk '{Size=$3 - $2}Size>50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV."$s"coord.seg_duplication.bed &

wait

#checking compliment structural variants against specific types of structural variants with kestrel coordiantes
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.deletion50.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV.kestCoord.full-v-deletion50.bed &
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.insertion50.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV.kestCoord.full-v-insertion50.bed &
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.inversion.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV.kestCoord.full-v-inversion.bed &
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_deletion.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV.kestCoord.full-v-seg_deletion.bed &
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_duplication.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV.kestCoord.full-v-seg_duplication.bed &

#checking compliment structural variants against specific types of structural variants with $s coordinates
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV."$s"coord.deletion50.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV."$s"coord.full-v-deletion50.bed &
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b  /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV."$s"coord.insertion50.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV."$s"coord.full-v-insertion50.bed &
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV."$s"coord.inversion.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV."$s"coord.full-v-inversion.bed &
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV."$s"coord.seg_deletion.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV."$s"coord.full-v-deletion.bed &
bedtools intersect -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV."$s"coord.seg_duplication.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/SV-Beds/"$s"-v-Kestrel100KBhap.SV."$s"coord.full-v-seg_duplication.bed &
wait
done

function convert2selfcoord {
INFILE=$1
SAMPLE=$2
OUTFILE=`echo $INFILE | perl -pe "s#kestCoord#$SAMPLE\_coord#" | perl -pe "s#_coord#coord#"`
TYPE=`echo $INFILE | perl -pe 's#.*kestCoord\.([a-zA-Z_]+)\d*\.uniq.*#$1#'`

#Deletions in $s coordinates
if [ $TYPE == deletion ]
then
KEST_COORDS=`cat $INFILE | perl -pe 's#\t#__#g'`
rm $OUTFILE
for p in $KEST_COORDS
do
KEST_SCAFF=`echo $p | perl -pe 's#__#\t#g' | awk '{print $1}'`
KEST_POS1=`echo $p | perl -pe 's#__#\t#g' | awk '{print $2}'`
KEST_POS2=`echo $p | perl -pe 's#__#\t#g' | awk '{print $3}'`
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$SAMPLE"-v-Kestrel100KBhap.coords | awk '$15~/_1/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $2<=kest_pos1{QScaff=$15; QPos1=$4; next}$14==kest_scaff && $1>=kest_pos2 && $15!=QScaff{exit}$14==kest_scaff && $1>=kest_pos2 && $15==QScaff{QPos2=$3}QPos1<QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos1 OFS QPos2; exit}QPos1>QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos2 OFS QPos1; exit}' >> $OUTFILE
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$SAMPLE"-v-Kestrel100KBhap.coords | awk '$15~/_2/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $2<=kest_pos1{QScaff=$15; QPos1=$4; next}$14==kest_scaff && $1>=kest_pos2 && $15!=QScaff{exit}$14==kest_scaff && $1>=kest_pos2 && $15==QScaff{QPos2=$3}QPos1<QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos1 OFS QPos2; exit}QPos1>QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos2 OFS QPos1; exit}' >> $OUTFILE
done
fi

#Insertions in $s coordinates
if [ $TYPE == insertion ]
then
KEST_COORDS=`cat $INFILE | perl -pe 's#\t#__#g'`
rm $OUTFILE
for p in $KEST_COORDS
do
KEST_SCAFF=`echo $p | perl -pe 's#__#\t#g' | awk '{print $1}'`
KEST_POS1=`echo $p | perl -pe 's#__#\t#g' | awk '{print $2}'`
KEST_POS2=`echo $p | perl -pe 's#__#\t#g' | awk '{print $3}'`
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$SAMPLE"-v-Kestrel100KBhap.coords | awk '$15~/_1/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $2<=kest_pos1{QScaff=$15; QPos1=$4; next}$14==kest_scaff && $1>=kest_pos2 && $15!=QScaff{exit}$14==kest_scaff && $1>=kest_pos2 && $15==QScaff{QPos2=$3}QPos1<QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos1 OFS QPos2; exit}QPos1>QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos2 OFS QPos1; exit}' >> $OUTFILE
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$SAMPLE"-v-Kestrel100KBhap.coords | awk '$15~/_2/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $2<=kest_pos1{QScaff=$15; QPos1=$4; next}$14==kest_scaff && $1>=kest_pos2 && $15!=QScaff{exit}$14==kest_scaff && $1>=kest_pos2 && $15==QScaff{QPos2=$3}QPos1<QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos1 OFS QPos2; exit}QPos1>QPos2 && QPos1!="" && QPos2!=""{print QScaff OFS QPos2 OFS QPos1; exit}' >> $OUTFILE
done
fi

if [ $TYPE == inversion ]
then
KEST_COORDS=`cat $INFILE | perl -pe 's#\t#__#g'`
rm $OUTFILE
for p in $KEST_COORDS
do
KEST_SCAFF=`echo $p | perl -pe 's#__#\t#g' | awk '{print $1}'`
KEST_POS1=`echo $p | perl -pe 's#__#\t#g' | awk '{print $2}'`
KEST_POS2=`echo $p | perl -pe 's#__#\t#g' | awk '{print $3}'`
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$SAMPLE"-v-Kestrel100KBhap.coords | awk '$15~/_1/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $1==kest_pos1 && $2==kest_pos2{QScaff=$15; QPos1=$3; QPos2=$4; print QScaff OFS QPos1 OFS QPos2}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' >> "$OUTFILE".temp
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$SAMPLE"-v-Kestrel100KBhap.coords | awk '$15~/_2/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $1==kest_pos1 && $2==kest_pos2{QScaff=$15; QPos1=$3; QPos2=$4; print QScaff OFS QPos1 OFS QPos2}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' >> "$OUTFILE".temp
done
cat "$OUTFILE".temp | sort -k1,1V -k2,2n | uniq > "$OUTFILE"
rm "$OUTFILE".temp
fi

if [ $TYPE == seg_deletion ]
then
KEST_COORDS=`cat $INFILE | perl -pe 's#\t#__#g'`
rm $OUTFILE
for p in $KEST_COORDS
do
KEST_SCAFF=`echo $p | perl -pe 's#__#\t#g' | awk '{print $1}'`
KEST_POS1=`echo $p | perl -pe 's#__#\t#g' | awk '{print $2}'`
KEST_POS2=`echo $p | perl -pe 's#__#\t#g' | awk '{print $3}'`
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$SAMPLE"-v-Kestrel100KBhap.coords | awk '$15~/_1/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $1==kest_pos1 && $2==kest_pos2{QScaff=$15; QPos1=$3; QPos2=$4; print QScaff OFS QPos1 OFS QPos2}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' >> "$OUTFILE".temp
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$SAMPLE"-v-Kestrel100KBhap.coords | awk '$15~/_2/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $1==kest_pos1 && $2==kest_pos2{QScaff=$15; QPos1=$3; QPos2=$4; print QScaff OFS QPos1 OFS QPos2}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' >> "$OUTFILE".temp
done
cat "$OUTFILE".temp | sort -k1,1V -k2,2n | uniq > "$OUTFILE"
rm "$OUTFILE".temp
fi

if [ $TYPE == seg_duplication ]
then
KEST_COORDS=`cat $INFILE | perl -pe 's#\t#__#g'`
rm $OUTFILE
for p in $KEST_COORDS
do
KEST_SCAFF=`echo $p | perl -pe 's#__#\t#g' | awk '{print $1}'`
KEST_POS1=`echo $p | perl -pe 's#__#\t#g' | awk '{print $2}'`
KEST_POS2=`echo $p | perl -pe 's#__#\t#g' | awk '{print $3}'`
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$SAMPLE"-v-Kestrel100KBhap.coords | awk '$15~/_1/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $1==kest_pos1 && $2==kest_pos2{QScaff=$15; QPos1=$3; QPos2=$4; print QScaff OFS QPos1 OFS QPos2}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' >> "$OUTFILE".temp
grep -P "^[0-9]+" /scratch/jw5478/FalconProject/Projects/RefGenome8/NucVar/Comparative/Kestrel_Alignments/MUMmerOutputs/"$SAMPLE"-v-Kestrel100KBhap.coords | awk '$15~/_2/{print $0}' | awk -v kest_scaff=$KEST_SCAFF -v kest_pos1=$KEST_POS1 -v kest_pos2=$KEST_POS2 'BEGIN{OFS=FS="\t"}$14==kest_scaff && $1==kest_pos1 && $2==kest_pos2{QScaff=$15; QPos1=$3; QPos2=$4; print QScaff OFS QPos1 OFS QPos2}' | awk 'BEGIN{OFS=FS="\t"}$2<$3{print $0}$2>$3{print $1 OFS $3 OFS $2}' >> "$OUTFILE".temp
done
cat "$OUTFILE".temp | sort -k1,1V -k2,2n | uniq > "$OUTFILE"
rm "$OUTFILE".temp
fi
}

for s in A D B FpBSx77 C E F Kestrel
do
DELETIONS=`ls /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/*/Variants/*-v-Kestrel100KBhap.SV.kestCoord.deletion50.bed | grep -v "/$s/" | grep -v "$s"-v-Kestrel100KBhap.SV.kestCoord.deletion50.bed`
INSERTIONS=`ls /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/*/Variants/*-v-Kestrel100KBhap.SV.kestCoord.insertion50.bed | grep -v "/$s/" | grep -v "$s"-v-Kestrel100KBhap.SV.kestCoord.insertion50.bed`
INVERSIONS=`ls /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/*/Variants/*-v-Kestrel100KBhap.SV.kestCoord.inversion.bed | grep -v "/$s/" | grep -v "$s"-v-Kestrel100KBhap.SV.kestCoord.inversion.bed`
SEG_DELETIONS=`ls /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/*/Variants/*-v-Kestrel100KBhap.SV.kestCoord.seg_deletion.bed | grep -v "/$s/" | grep -v "$s"-v-Kestrel100KBhap.SV.kestCoord.seg_deletion.bed`
SEG_DUPLICATIONS=`ls /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/*/Variants/*-v-Kestrel100KBhap.SV.kestCoord.seg_duplication.bed | grep -v "/$s/" | grep -v "$s"-v-Kestrel100KBhap.SV.kestCoord.seg_duplication.bed`

cat /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/*/Variants/*-v-Kestrel100KBhap.SV.kestCoord.deletion50.bed | bedtools subtract -A -a - -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.deletion50.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV.kestCoord.deletion50.strip.bed &
cat /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/*/Variants/*-v-Kestrel100KBhap.SV.kestCoord.insertion50.bed | bedtools subtract -A -a - -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.insertion50.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV.kestCoord.insertion50.strip.bed &
cat /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/*/Variants/*-v-Kestrel100KBhap.SV.kestCoord.inversion.bed | bedtools subtract -A -a - -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.inversion.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV.kestCoord.inversion.strip.bed &
cat /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/*/Variants/*-v-Kestrel100KBhap.SV.kestCoord.seg_deletion.bed | bedtools subtract -A -a - -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_deletion.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_deletion.strip.bed &
cat /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/*/Variants/*-v-Kestrel100KBhap.SV.kestCoord.seg_duplication.bed | bedtools subtract -A -a - -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_duplication.bed > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_duplication.strip.bed &

bedtools subtract -A -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.deletion50.bed -b $DELETIONS > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV.kestCoord.deletion50.uniq.bed &
bedtools subtract -A -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.insertion50.bed -b $INSERTIONS  > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV.kestCoord.insertion50.uniq.bed &
bedtools subtract -A -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.inversion.bed -b $INVERSIONS  > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV.kestCoord.inversion.uniq.bed  &
bedtools subtract -A -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_deletion.bed -b $SEG_DELETIONS > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_deletion.uniq.bed &
bedtools subtract -A -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_duplication.bed -b $SEG_DUPLICATIONS > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV.kestCoord.seg_duplication.uniq.bed &
wait

LIST_FILES=`ls /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV.kestCoord.*.uniq.bed`
for f in $LIST_FILES
do
convert2selfcoord $f $s  &
done
wait

for t in deletion50 insertion50 inversion seg_deletion seg_duplication
do
bedtools intersect -F 1 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-rep.bed &
bedtools intersect -F 1 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.simp.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-simp.bed &
bedtools intersect -F 1 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.DNA.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-DNA.bed &
bedtools intersect -F 1 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.LTR.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-LTR.bed &
bedtools intersect -F 1 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.LINE.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-LINE.bed &
bedtools intersect -F 1 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.SINE.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-SINE.bed &
bedtools intersect -F 1 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.unknown.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-unknown.bed &

bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-rep50.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.simp.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-simp50.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.DNA.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-DNA50.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.LTR.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-LTR50.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.LINE.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-LINE50.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.SINE.bed | awk '{Size=$3 - $2}Size=5>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-SINE50.bed &
bedtools intersect -F 1 -f 0.5 -a /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Variants/Uniques/"$s"-v-Kestrel100KBhap.SV."$s"coord."$t".uniq.bed -b /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/"$s"/RepeatBeds/"$s"_16KBph2.diploid.fasta.rep.unknown.bed | awk '{Size=$3 - $2}Size>=50{print $0}' > /scratch/jw5478/FalconProject/Projects/RefGenome8/Repeat-SV/SV/KestrelAlign/"$s"/Rep-SVBeds/"$s"_Kestrel100KBhap."$s"coord."$t"-unknown50.bed &
wait
done
done
